### DISCUSSION TITLE
Support access tokens for specific packages

### DISCUSSION THREAD

**Discussion by SimenB** [npm/feedback#636] [Registry] [answered]
I'd love to generate tokens that only work for a single package, or a set of packages. When I generate a token today, that token has access to e.g. all of Jest, which I feel slightly uneasy about. I'd love to be able to scope any generated token to specific packages.

(also, thank you for finally allowing us to name tokens!)

**Comment by MylesBorins** [comment, marked as answer]
It's something we want to do and it is [on our public roadmap](https://github.com/npm/roadmap/issues/10)

We can't commit to any dates right now but it is currently being discussed!

**Comment by SimenB** [comment]
Oh, I thought I'd searched for it. Thanks for the link!

EDIT: Ah, but I searched GH's roadmap, not npm's 😅

**Reply by MylesBorins** [reply]
we are working on improving this :D
